arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt

```diff
@@ -146,4 +147,13 @@ fun <K, V> genMap(genK: Gen<K>, genV: Gen<V>): Gen<Map<K, V>> =
         Gen.create { Gen.list(genK).generate().map { it to genV.generate() }.toMap() }
 
 fun <K, V> genMapKW(genK: Gen<K>, genV: Gen<V>): Gen<MapKW<K, V>> =
-        Gen.create { Gen.list(genK).generate().map { it to genV.generate() }.toMap().k() }
\ No newline at end of file
+        Gen.create { Gen.list(genK).generate().map { it to genV.generate() }.toMap().k() }
+
+class TimeUnitGen : Gen<TimeUnit> {
+
+    companion object {
+        val units = TimeUnit.values()
+    }
+
+    override fun generate(): TimeUnit = units[Gen.choose(0, units.size -1).generate()]
```

Comment:
A new Gen.choose() is created on every generate call. This makes the seed to be reset every time, which messes with the entropy. You should have it as a field of the class instead.